crates/nu-parser/src/parser.rs

```diff
@@ -3291,8 +3294,22 @@ pub fn parse_signature_helper(
                 } else {
                     match parse_mode {
                         ParseMode::ArgMode => {
+                            macro_rules! valid_variable_name {
+                                ($name:expr) => {
+                                    if !is_variable($name) {
+                                        error = error.or_else(|| {
+                                            Some(ParseError::Expected(
+                                                "valid variable name".into(),
+                                                span,
+                                            ))
+                                        })
+                                    }
+                                };
```

Comment:
Not convinced that this macro is necessary for this few usages. I would rather have a function that emits a useful error and has documentation to encourage usage in other places if the need arises.

Reply:
I don't know how you'd rewrite this to be a function given that the point of it is to dedup the `error = error.or_else()` code structure, not just produce the `ParseError::Expected`. I also think 6 uses is about enough for an 8-line block. But w/e.